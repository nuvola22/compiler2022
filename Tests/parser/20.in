var a: Real = 10;
    b: Real = 11;
    c: Real = 12;
    d: Real = 13;

begin
    writeln(+3);
    writeln(-3);
    writeln(-+---++--3);
    writeln(2+++++3);
    writeln(2-++++3);

    writeln(2 + 3);
    writeln(2 - 3);
    writeln(2 * 3);
    writeln(2 / 3);

    writeln(a + 3);
    writeln(a - 3);
    writeln(a * 3);
    writeln(a / 3);

    writeln(a + a);
    writeln(a - a);
    writeln(a * a);
    writeln(a / a);

    writeln(a + b + c + d);
    writeln(a + b + c + d * a);

    writeln(a - b - c - d);
    writeln(a - b - c - d * a);

    writeln(a / b * c / b);

    writeln(a div b);

end.