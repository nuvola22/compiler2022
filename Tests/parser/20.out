      program
         variable
            a
            Real
            10
         variable
            b
            Real
            11
         variable
            c
            Real
            12
         variable
            d
            Real
            13
         begin
            call
               writeln
               +
                  Parser.NodeNumber
            call
               writeln
               -
                  Parser.NodeNumber
            call
               writeln
               -
                  Parser.NodeUnOp
            call
               writeln
               +
                  2
                  +
                     Parser.NodeUnOp
            call
               writeln
               -
                  2
                  +
                     Parser.NodeUnOp
            call
               writeln
               +
                  2
                  3
            call
               writeln
               -
                  2
                  3
            call
               writeln
               *
                  2
                  3
            call
               writeln
               /
                  2
                  3
            call
               writeln
               +
                  a
                  3
            call
               writeln
               -
                  a
                  3
            call
               writeln
               *
                  a
                  3
            call
               writeln
               /
                  a
                  3
            call
               writeln
               +
                  a
                  a
            call
               writeln
               -
                  a
                  a
            call
               writeln
               *
                  a
                  a
            call
               writeln
               /
                  a
                  a
            call
               writeln
               +
                  +
                     +
                        a
                        b
                     c
                  d
            call
               writeln
               +
                  +
                     +
                        a
                        b
                     c
                  *
                     d
                     a
            call
               writeln
               -
                  -
                     -
                        a
                        b
                     c
                  d
            call
               writeln
               -
                  -
                     -
                        a
                        b
                     c
                  *
                     d
                     a
            call
               writeln
               /
                  *
                     /
                        a
                        b
                     c
                  b
            call
               writeln
               div
                  a
                  b
